static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js

```diff
@@ -1,34 +1,95 @@
 define([
-    'lib/fetch-json'
+    'Promise',
+    'lib/fetch-json',
+    'lib/cookies',
+    'lib/storage',
+    'lib/report-error'
 ], function(
-    fetchJson
+    Promise,
+    fetchJson,
+    cookies,
+    storage,
+    reportError
 ) {
-    function getEmail(browserId) {
-        return fetchJson('https://tailor.guardianapis.com/email/' + browserId +'?emailIds=39,3322,3039,1950,38,3698', {
-            method: 'get'
-        });
+
+    var URLS = {
+        suggestions: 'https://tailor.guardianapis.com/suggestions?browserId='
+    };
+
+    var browserId = cookies.get('bwid');
+
+    function getURL(type, queryParams) {
+        var baseURL = URLS[type];
+
+        if (!baseURL) {
+            return '';
+        }
+
+        baseURL += browserId;
+
+        if (queryParams) {
+            var queryParamList = Object.keys(queryParams);
+
+            baseURL += '?';
+
+            queryParamList.forEach(function (queryParam, i) {
+                baseURL += queryParam + '=' + queryParams[queryParam];
+
+                if (i < (queryParamList.length - 1)) {
+                    baseURL += '&';
+                }
+            });
+        }
+
+        return baseURL;
     }
 
-    function getSuggestions(browserId) {
-        return fetchJson('https://tailor.guardianapis.com/suggestions?browserId=' + browserId, {
-            method: 'get'
-        });
+    /**
+     * type (required) is a string which should match a key in the URLS object
+     * queryParams (optional) is an object literal, each key/value will be query string parameter
+     * eg. {foo:'bar', hello:'world'} translates to ?foo=bar&hello=world
+     *
+    **/
+    function fetchData(type, queryParams) {
+        var tailorData = storage.local.get('gu.tailor.' + type);
+
+        if (tailorData) {
+            return Promise.resolve(tailorData);
+        }
+
+        if (!browserId) {
+            return Promise.resolve({});
+        }
+
+        var url = getURL(type, queryParams);
+
+        if (!url) {
+            return Promise.resolve({});
+        }
+
+        return fetchJson(url, {
+                    method: 'get'
+                })
+                .then(handleResponse.bind(null, type))
+                .catch(handleError.bind(null, type))
     }
 
-    function getRegularStatus(browserID) {
-        return getSuggestions(browserID).then(function(res) {
-            try {
-                return res.userDataForClient.regular;
-            } catch (e) {
-                return false
-            }
-        })
+    function handleResponse(type, data) {
+        var day = 1000 * 60 * 60 * 24;
+
+        storage.local.set('gu.tailor.' + type, data, { expires: Date.now() + day });
```

Comment:
Do we need to worry about checking the existence of local storage for a browser? Or is this handled in the storage module?